src/adagrams.js

```diff
@@ -1,15 +1,154 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+const LETTER_POOL = {
+  'A': 9, 
+  'B': 2, 
+  'C': 2, 
+  'D': 4, 
+  'E': 12, 
+  'F': 2, 
+  'G': 3, 
+  'H': 2, 
+  'I': 9, 
+  'J': 1, 
+  'K': 1, 
+  'L': 4, 
+  'M': 2, 
+  'N': 6, 
+  'O': 8, 
+  'P': 2, 
+  'Q': 1, 
+  'R': 6, 
+  'S': 4, 
+  'T': 6, 
+  'U': 4, 
+  'V': 2, 
+  'W': 2, 
+  'X': 1, 
+  'Y': 2, 
+  'Z': 1
+};
+
+const LETTER_SCORE = {
+  'A': 1, 
+  'B': 3, 
+  'C': 3, 
+  'D': 2, 
+  'E': 1, 
+  'F': 4, 
+  'G': 2, 
+  'H': 4, 
+  'I': 1, 
+  'J': 8, 
+  'K': 5, 
+  'L': 1, 
+  'M': 3, 
+  'N': 1, 
+  'O': 1, 
+  'P': 3, 
+  'Q': 10, 
+  'R': 1, 
+  'S': 1, 
+  'T': 1, 
+  'U': 1, 
+  'V': 4, 
+  'W': 4, 
+  'X': 8, 
+  'Y': 4, 
+  'Z': 10
 };
 
+const HAND_SIZE = 10
+const MIN_BONUS_LENGTH = 7
+const LENGTH_BONUS = 8
+
+const createWeightedLetterPool = () =>{
+  let weightedLetterPool = [];
+  for (const [letter, freq] of Object.entries(LETTER_POOL)){
+    for (let i=0; i<freq; i++){
+      weightedLetterPool.push(letter);
+    }
+  }
+  return weightedLetterPool;
+}
+
+export const drawLetters = () => {
+  const hand = [];
+  const weightedLetterPool = createWeightedLetterPool()
+
+  while (hand.length < HAND_SIZE){
+    let randomIdx = Math.floor(Math.random()*weightedLetterPool.length);
+    let randomLetter = weightedLetterPool[randomIdx];
+    hand.push(randomLetter);
+    weightedLetterPool.splice(randomIdx,1);
+  }
+  return hand;
+};
+  
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const inputLower = input.toUpperCase();
+  const copy = [...lettersInHand]; // I not sure how lettersInHand is used in the program as a whole, so I made a copy
+
+  for (let letter of inputLower){
+    if (copy.includes(letter)){
+      let idx = copy.indexOf(letter);
+      copy.splice(idx,1);
+    } else {
+      return false;
+    }
+  }
+  return true;
 };
 
+  
+
+
+
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  if (!word.trim()) {
+    return 0;
+  }
```

Comment:
I would leave off this trim behavior. This is granting special status to spaces at the start or end of the word (without granting the same status to spaces within the word). Personally, I'm not even a fan of this function needing to deal with case (though that's a test requirement). I would see this function as scoring a valid word, which should already have accounted for any invalid tiles or vagueries of case before getting here.